MainWindow.cpp

```diff
@@ -155,7 +188,7 @@ MainWindow::MainWindow(QWidget *parent) : QMainWindow(parent), _ui(new Ui::MainW
   openNewProject();
 }
 
-MainWindow::~MainWindow() { delete _ui; }
+MainWindow::~MainWindow() { diagnosticTextEdit = nullptr; delete _ui; }

```

Comment:
This isn't necessary it uses the Qt ownership mechanism (it's parent is specified in the constructor). This is C++, not C, we're not concerned about zero'ing out our pointers.

https://doc.qt.io/qt-5/objecttrees.html

Reply:
The editor segfaults on close without this. You're referencing another pointer so it's never cleared.

Reply:
I think it's the signals queueing messages, something else is wrong, this is only a bandaid. The correct fix is to disconnect the output/log signals so it stops trying to reference that after it's gone.

Reply:
it also correct to null the pointer